src/localization/LanguageSwitcher.tsx

```diff
@@ -0,0 +1,79 @@
+import { type ReactNode, createContext, useContext, useState } from "react";
```

Comment:
I've not come across a LanguageSwitcher before, is this actually needed I don't see anything like this in OSS

Reply:
Didn't realize that we'll reference OSS on this project, sorry. But this was actually just a "project name" for the file, did think in the beginning that it's not a good name for file but somehow ended up keeping it... Will rename appropriately.

Reply:
I could be mistaken but I believe this whole file can be removed and is not necessary, we just use the localization package to achieve what we need. We can check this together though